src/axom/core/Array.hpp

```diff
@@ -667,15 +691,15 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
    */
   virtual void dynamicRealloc(IndexType new_num_elements);
 
-  T* m_data = nullptr;
+  T* m_data {nullptr};
```

Comment:
Are changes like this going to introduce confusion in coding style? What is the benefit of doing it this way vs. '= val'?

Reply:
strongly agree with Rich's comment

Reply:
I agree about consistency and am not aware of any semantic differences.
I'll revert this.

On a related note, perhaps we should run the code through clang-tidy (with an appropriate set of agreed upon rules).

Reply:
especially for built-in types, I think it's less jarring with the common '=' syntax that everyone uses.

Reply:
But, nice effort trying to sneak one past the goalie.... 😆 I know @agcapps would have said something if I didn't.

Reply:
I sent kenny a teams message about this very thing and he straight up ignored me! I'm deeply offended! (I just assumed it was some weird C++ optimization thing... but still...)

Reply:
There are some subtle differences, according to [this handy table](https://josuttis.com/cpp/c++initialization.pdf)

It would seem the main benefit of using braces would be to detect unintended implicit conversions, although braces do complicate things when it comes to types with initializer list constructors.

Reply:
hey, who is the white238 person crashing our party?!?

Reply:
@kanye-quest thanks for the pointer to the "handy table"

Reply:
/sad

am i not allowed in code anymore?

/goes back to build and spack land......

Reply:
@white238 i think the issue is that you're mixing communication mediums too much, like a couple of days ago when you responded to my email via teams  😄

Reply:
its to keep you guys guessing.. where will white238 be next! (hint probably teams)

Reply:
I reverted the changes to `axom::Array`

Thanks for the link @kanye-quest. 
fwiw: The `Type i {val}` form of the expression has checks across the board, so it might be the best?

@white238 -- looking at our chat, I responded that the two should be equivalent! But there were other discussions, so it might have gotten lost.